### PR TITLE
ZCS-11909 Error is displaying for OpenIO in zmvolume -l

### PR DESCRIPTION
**Pull request by inpati**
**Problem Statement**
Error is displaying for OpenIO in zmvolume -l, getAllVolumeRequest is failing for Open IO volumes.

**Solution**
Fixed null issue for external volumes while calling zmvolume -l

### CODE REVIEW

**Review by zimsuchitgupta — APPROVED** [review]

File: store/src/java/com/zimbra/cs/volume/VolumeCLI.java [hunk 1]
```diff
@@ -597,4 +606,17 @@ private String toTypeName(short type) {
...
         return "Unrecognized type " + type;
     }
-}
\ No newline at end of file
+
+    /**
+     * Returns the storage type name for the given volume info type.
+     */
+    private String extractStorageType(VolumeInfo volInfo) {
+        String result = "";
```
Comment (vishnukantjangid) [inline store/src/java/com/zimbra/cs/volume/VolumeCLI.java:614]: Here, we don't need to initialize with blank. Initialize with null only

File: store/src/java/com/zimbra/cs/volume/VolumeCLI.java [hunk 2]
```diff
@@ -104,7 +104,8 @@ public final class VolumeCLI extends SoapCLI {
     private static final String A_BUCKET_ID = "bucketId";
     private static final String A_VOLUME_PREFIX = "volumePrefix";
     private static final String A_VOLUME_S3 = "S3";
-    
+    private static final String A_VOLUME_OPEN_IO = "OPENIO";
```
Comment (RakeshAMore) [inline store/src/java/com/zimbra/cs/volume/VolumeCLI.java:107]: Check if we have a constant for it and use that and make sure this is the same value that we are accepting creating volume.

File: store/src/java/com/zimbra/cs/volume/VolumeCLI.java [hunk 3]
```diff
@@ -256,27 +257,35 @@ private void print(VolumeInfo vol) {
...
         if (vol.isCompressBlobs()) {
             System.out.println("\t                 threshold: " + vol.getCompressionThreshold() + " bytes");
         } else {
             System.out.println();
         }
         System.out.println("                     current: " + vol.isCurrent());
-        if(vol.getStoreType() == 2) {
-            VolumeExternalInfo volumeExternalInfo = vol.getVolumeExternalInfo();
-            if (A_VOLUME_S3.equals(volumeExternalInfo.getStorageType())) {
+        if (vol.getStoreType() == 2) {
```
Comment (RakeshAMore) [inline store/src/java/com/zimbra/cs/volume/VolumeCLI.java:267]: We have constant/enum for `2` right?

File: store/src/java/com/zimbra/cs/volume/VolumeCLI.java [hunk 4]
```diff
@@ -597,4 +606,17 @@ private String toTypeName(short type) {
...
-}
\ No newline at end of file
+
+    /**
+     * Returns the storage type name for the given volume info type.
+     */
+    private String extractStorageType(VolumeInfo volInfo) {
+        String result = "";
+        if (null != volInfo.getVolumeExternalInfo()) {
+            result = volInfo.getVolumeExternalInfo().getStorageType();
```
Comment (RakeshAMore) [inline store/src/java/com/zimbra/cs/volume/VolumeCLI.java:616]: This seems to be our tech debt, we will revisit this.

File: store/src/java/com/zimbra/cs/volume/VolumeCLI.java (hunk 3 above)
Comment (inpati) [reply store/src/java/com/zimbra/cs/volume/VolumeCLI.java:267]: We don't have any constant enum for 2 or 1, we were going to write in new repo but we didnt had that conts

File: store/src/java/com/zimbra/cs/volume/VolumeCLI.java (hunk 1 above)
Comment (inpati) [reply store/src/java/com/zimbra/cs/volume/VolumeCLI.java:614]: ok fixed

File: store/src/java/com/zimbra/cs/volume/VolumeCLI.java (hunk 4 above)
Comment (inpati) [reply store/src/java/com/zimbra/cs/volume/VolumeCLI.java:616]: Yes, as I dint wanted to touch vol-info class managed here in volumecli class with small function

File: store/src/java/com/zimbra/cs/volume/VolumeCLI.java (hunk 2 above)
Comment (inpati) [reply store/src/java/com/zimbra/cs/volume/VolumeCLI.java]: Used Admin constants for A_VOLUME_OPEN_IO and A_VOLUME_S3

File: store/src/java/com/zimbra/cs/volume/VolumeCLI.java (hunk 3 above)
Comment (RakeshAMore) [reply store/src/java/com/zimbra/cs/volume/VolumeCLI.java:266]: see this https://github.com/Zimbra/zm-mailbox/blob/feature/9.1.0/store/src/java/com/zimbra/cs/volume/Volume.java#L71

Comment (inpati) [reply store/src/java/com/zimbra/cs/volume/VolumeCLI.java:266]: Ok. fixed

**Review by dasiyogesh — APPROVED** [review]

**Review by RakeshAMore — APPROVED** [review]

**Pull request merged**